rasterio/tools/mask.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import absolute_import
+
+import rasterio
+from rasterio.features import geometry_mask
+
+def mask(raster, shapes, nodatavals=None, crop=False, all_touched=False,
+         invert=False):
+    """
+    For all regions in the input raster outside of the regions defined by
+    `shapes`, sets any data present to nodata.
+
+    Parameters
+    ----------
+    raster: rasterio RasterReader object
+        Raster to which the mask will be applied.
+    shapes: list of polygons
+        Polygons are GeoJSON-like dicts specifying the boundaries of features
+        in the raster to be kept. All data outside of specified polygons
+        will be set to nodata.
+    nodatavals: list (opt)
+        Value representing nodata within each raster band. If not set,
+        defaults to the nodatavals for the input raster. If those values
+        are not set, defaults to 0.
+    crop: bool (opt)
+        Whether to crop the raster to the extent of the data. Defaults to True.
+    all_touched: bool (opt)
+        Use all pixels touched by features. If False (default), use only
+        pixels whose center is within the polygon or that are selected by
+        Bresenhams line algorithm.
+    invert: bool (opt)
+        If True, mask will be True for pixels that overlap shapes.
+        False by default.
+
+    Returns
+    -------
+    masked: numpy ndarray
+        Data contained in raster after applying the mask.
+    out_transform: affine object
+        Information for mapping pixel coordinates in `masked` to another
+        coordinate system.
+    """
+
+    if crop and invert:
```

Comment:
@jdmcbr this should be handled this way only in `rio mask` because of the nature of combining CLI options.  Here you should raise a `ValueError` if both are `True`